docs/mutations.md

```diff
@@ -0,0 +1,48 @@
+# Mutate object after instantion
+
+The _Container_ supports objects to be mutated after it is constructed.
+
+After the _Container_ constructs a new instance of an object, you can specify which other objects will mutate the original object before locking it.
+
+Say we have classes like the following:
+
+```php
+namespace Vendor\Package;
+
+use Aura\Di\Injection\MutationInterface;
+
+class Example
+{
+    protected $foo;
+
+    public function setFoo($foo)
+    {
+        $this->foo = $foo;
+    }
+}
+
+class ExampleMutation implements MutationInterface
+{
+    public function mutate(object $object): object
+    {
+        $object->setFoo('mutated');
```

Comment:
There is a problem I think.  If the setters are relaying on other objects which have dependency on di then it will not work as expected right ? The mutation expects the setters are made possible via normal intantiation or string currently.

Reply:
A second option would be to pass the container as a second argument to the `mutate` method.

Reply:
Yes, I think it may be good to have container object.

Reply:
@harikt I had a look at it and my suggestion to pass the container as second argument is not possible after all. At least not without a large refactor because the container is not available in [`InjectionFactory`](https://github.com/auraphp/Aura.Di/blob/39ad3b89091556cc6687887d4be9a4f2366105c0/src/Injection/InjectionFactory.php#L92) and [`Factory`](https://github.com/auraphp/Aura.Di/blob/39ad3b89091556cc6687887d4be9a4f2366105c0/src/Injection/Factory.php#L122). So in order to pass the container to the `mutate` method we should somehow make the container available in those two classes (inject it).

I don't know if that is worth the effort since the mutation class can already be constructed as any other class. I added another example in the docs and included some more tests. The following works without a problem. I think this is in line with the rest of the library.

```php
$di->params['Vendor\Mutation']['argX'] = $di->lazyGet('service');
$di->params['Vendor\Mutation']['argY'] = $di;
$di->mutations['Vendor\Object'][] = $di->lazyNew('Vendor\Mutation');
```